ws-nextjs-app/pages/[service]/send/[id]/FormContext/utils/validateFunctions.ts

```diff
@@ -0,0 +1,98 @@
+import { FieldData, InvalidMessageCodes } from '../../types';
+
+const isStringEmpty = (str: string) =>
+  str == null || str.replaceAll(/\s/g, '').length <= 0;
+
+const isInvalidText: (data: FieldData) => FieldData = (data: FieldData) => {
+  const { required, value } = data;
+
+  let messageCode = null;
+  let isInvalid = false;
+  const isInputEmpty = isStringEmpty(value as string);
+
+  if (required && isInputEmpty) {
+    messageCode = InvalidMessageCodes.FieldRequired;
+    isInvalid = true;
+  }
+
+  return { ...data, invalid: isInvalid, messageCode };
+};
+
+const isInvalidEmail: (data: FieldData) => FieldData = (data: FieldData) => {
```

Comment:
Do we need to validate email address beyond what the browser gives us automatically? 

The `<input type="email />` element should give us some validation for free: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#validation. 

The docs correctly point out that its basic and more may be required, but the UGC API should help us a bit here.

Just trying to think of the minimum we can get away with without making it too complex.

Reply:
After chatting with @shayneahchoon we'll keep these manual validation functions as it allows for custom UI when validation fails.